File: hw/chalan/u1/hw03ChickenFarmSimulator/Chiken.py
from Poop import Poop
from Egg import Egg

class Chiken:
    def __init__(self, id: int, name: str, color: str, age: int, is_molting: bool):
        self.id = id
        self.name = name
        self.color = color
        self.age = age
        self.is_molting = is_molting

    def __str__(self):
        return (f"\nChiken{{\nid---->\t{self.id}, "
                f"\nname---->\t{self.name}, "
                f"\ncolor-->\t{self.color}, "
                f"\nage---->\t{self.age}, "
                f"\nisMolting--->{self.is_molting}}}")

    def _cluck(self):
        print(f"Chicken {self.name} is clucking cluck cluck cluck")

    def _eat(self):
        print(f"Chicken {self.name} is eating, grains")

    def poop(self, amount: int) -> Poop:
        poop = Poop(amount)
        print(f"Chicken {self.name} is pooping {poop}")
        return poop

    def lay_an_egg(self, size: str) -> Egg:
        egg = Egg(size)
        print(f"Chicken {self.name} is laying {egg.size} size egg")
        return egg

    def drink(self):
        print(f"Chicken {self.name} is drinking")

    def wander(self):
        print(f"Chicken {self.name} is wandering")

    def do_stuff(self):
        self._cluck()
        self._eat()
        self._cluck()
        self.poop(2)
        self.poop(3)
        self._eat()
        self.wander()
        self.drink()
        self.lay_an_egg('M')
        self.lay_an_egg('L')

File: hw/chalan/u1/hw03ChickenFarmSimulator/Chiken.js
const Poop = require('./Poop');
const Egg = require('./Egg');

class Chiken {
    constructor(id, name, color, age, isMolting) {
        this.id = id;
        this.name = name;
        this.color = color;
        this.age = age;
        this.isMolting = isMolting;
    }

    toString() {
        return `\nChiken{
id---->\t${this.id},
name---->\t${this.name},
color-->\t${this.color},
age---->\t${this.age},
isMolting--->${this.isMolting}}`;
    }

    // Métodos privados (convención _)
    _cluck() {
        console.log(`Chicken ${this.name} is clucking cluck cluck cluck`);
    }

    _eat() {
        console.log(`Chicken ${this.name} is eating, grains`);
    }

    poop(amount) {
        const poop = new Poop(amount);
        console.log(`Chicken ${this.name} is pooping ${poop.toString()}`);
        return poop;
    }

    layAnEgg(size) {
        const egg = new Egg(size);
        console.log(`Chicken ${this.name} is laying ${egg.size} size egg`);
        return egg;
    }

    drink() {
        console.log(`Chicken ${this.name} is drinking`);
    }

    wander() {
        console.log(`Chicken ${this.name} is wandering`);
    }

    doStuff() {
        this._cluck();
        this._eat();
        this._cluck();
        this.poop(2);
        this.poop(3);
        this._eat();
        this.wander();
        this.drink();
        this.layAnEgg('M');
        this.layAnEgg('L');
    }
}

module.exports = Chiken;
